source/adapters/level_zero/CMakeLists.txt

```diff
@@ -6,6 +6,19 @@
 
 set(TARGET_NAME ur_adapter_level_zero)
 
+# Copy L0 loader/headers locally to the build to avoid leaking their path.
```

Comment:
This looks to me like we are externalizing details of a specific CI pipeline. Given that you can already specify a path to level zero, would it be possible for this copying step to happen outside of the CMake scripts as one of the CI steps? And then you can point to this location using the `LEVEL_ZERO_LIBRARY` and `LEVEL_ZERO_INCLUDE_DIR` variables.

Reply:
Thanks @pbalcer . Yes, doing it in CI is OK, however, we would still leak path if we build without CI.

Reply:
Where exactly is the path being leak?

Reply:
eg: we are using nfs path to share the pre built L0 library/headers among different projects. If we don't copy them locally, we would leak the full path of nfs path .

Reply:
Right, but does the copying have to happen during the CMake configuration step? There must be some script/configuration file that triggers the build, which could look like this:
```
cp $L0_LIBRARY /tmp/level-zero/ # copy from NFS to a local directory
cmake ... -DLEVEL_ZERO_LIBRARY=/tmp/level-zero/ # configure unified runtime
```

Or am I misunderstanding something?

Reply:
Sorry, I should have been clearer. What's the mechanism by which someone might find this information about internal infrastructure?

Do these paths get embedded into the binary objects output by the build?

Are the paths in CI job logs which become public?

Or something else.

Reply:
> Do these paths get embedded into the binary objects output by the build?

Yes, the path might be leak in binaries, especially in PDB.

Reply:
So this is for compatibility with existing tooling that already uses these specific variable names. Sounds reasonable. I'd prefer to avoid special-casing such things in CMake scripts, but I guess merging this is the least disruptive approach.

Reply:
@kbenzie Any further questions? Thanks.

Reply:
None from me.